include/SFML/Graphics/Shader.hpp

```diff
@@ -457,6 +457,18 @@ class SFML_GRAPHICS_API Shader : GlResource, NonCopyable
     void setParameter(const std::string& name, CurrentTextureType);
 
     ////////////////////////////////////////////////////////////
+    /// \brief Get the underlying OpenGL handle of the shader.
+    ///
+    /// You shouldn't need to use this function, unless you have
+    /// very specific stuff to implement that SFML doesn't support,
+    /// or implement a temporary workaround until a bug is fixed.
+    ///
+    /// \return OpenGL handle of the shader or 0 if not yet loaded
+    ///
+    ////////////////////////////////////////////////////////////
+    unsigned int getNativeHandle() const;
```

Comment:
I'm wondering if the cast macros from `src/SFML/Graphics/Shader.cpp` should be documented here. Having never use those handle myself I don't know whether or not the user will have some trouble with Apple's special types...